packages/@aws-cdk/aws-cognito/lib/user-pool-domain.ts

```diff
@@ -126,4 +130,32 @@ export class UserPoolDomain extends Resource implements IUserPoolDomain {
     });
     return customResource.getResponseField('DomainDescription.CloudFrontDistribution');
   }
+
+  /**
+   * The URL to the hosted UI associated with this domain
+   */
+  public baseUrl(): string {
+    if (this.isCognitoDomain) {
+      return `https://${this.domainName}.auth.${Stack.of(this).region}.amazoncognito.com`;
+    }
+    return `https://${this.domainName}`;
+  }
+
+  /**
+   * The URL to the sign in page in this domain using a specific UserPoolClient
+   * @param client [disable-awslint:ref-via-interface] the user pool client that the UI will use to interact with the UserPool
+   * @param redirectUri the URI to redirect to after sign in. This URI must be one specified in the `callbackUrl`
+   * property under `oAuth` with the specified client.
+   */
+  public signInUrl(client: UserPoolClient, redirectUri: string): string {
+    let responseType: string;
+    if (client.oAuthFlows.authorizationCodeGrant) {
+      responseType = 'code';
+    } else if (client.oAuthFlows.implicitCodeGrant) {
+      responseType = 'token';
+    } else {
+      throw new Error('signInUrl is not supported for clients without authorizationCodeGrant or implicitCodeGrant flow enabled');
+    }
+    return `${this.baseUrl()}/login?client_id=${client.userPoolClientId}&response_type=${responseType}&redirect_uri=${redirectUri}`;
```

Comment:
The `/login` route should probably be configurable because custom domains may map a different route for this purpose.